CMSgov/design-system
# [WNMGDS-2928] Standardize `inputRef` prop types

## Summary

WNMGDS-2928

- Standardizes `inputRef` prop types to use the React `Ref` type, which covers both Object refs and ref functions
- Adds unit tests to make sure both kinds of refs work

## How to test

1. Run unit tests
2. Try one of the ChoiceList stories and verify that the `onComponentBlur` event still works

## Checklist


### If this is a change to code:
